argilla/src/argilla/datasets/_export/_hub.py

```diff
@@ -0,0 +1,194 @@
+# Copyright 2024-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import warnings
+from collections import defaultdict
+from tempfile import TemporaryDirectory
+from typing import Any, Type, TYPE_CHECKING, Optional, Union
+from uuid import UUID
+
+from argilla.records import Record
+from argilla.responses import Response
+from argilla._exceptions import NotFoundError
+from argilla._helpers import resolve_hf_datasets_type
+
+if TYPE_CHECKING:
+    from argilla import Dataset
+    from argilla import Argilla
+    from argilla import Workspace
+
+HFDataset = resolve_hf_datasets_type()
```

Comment:
For what I've seen, `HFDataset` is used only for typing. Is there a reason why not directly doing:

```py
if TYPE_CHECKING:
    from argilla import Dataset
    from argilla import Argilla
    from argilla import Workspace
    from datasets import Dataset as HFDataset
```
?

Since all the code inside `if TYPE_CHECKING` is never executed, you won't have an import issue even if `datasets` is not installed. And this way is more explicit than `resolve_hf_datasets_type` in my opinion (+ it avoids an import at runtime if `datasets` is not needed)

Reply:
We were avoiding having `datasets` as a dependency for the core `argilla` install. Instead, it's installed under a parameter `argilla[io]`. But I also question whether that still a good idea.

@frascuchon What do you think? Should we make datasets a core dependency and simplify this?

Reply:
To be confirmed but having `datasets` under `if TYPE_CHECKING` statement shouldn't make it required since it'll never be executed.

Reply:
Just to confirm, `huggingface_hub` is required by `datasets` but not the other way around.